ros2component/ros2component/api/__init__.py

```diff
@@ -82,7 +81,10 @@ def get_container_components_info(*, node, remote_container_node_name):
         '{}/_container/list_nodes'.format(remote_container_node_name)
     )
     try:
-        list_nodes_client.wait_for_service()
+        if not list_nodes_client.wait_for_service(timeout_sec=5.0):
```

Comment:
When used in a loop (as in `component list`) a blocking timeout of 5s can add up to a very long time for the user call.

Reply:
Fair enough. Timeout is a somewhat arbitrary, larger than zero quantity. What about `2.0` seconds?

Reply:
I don't mind the value. For this PR you can choose whatever.

The problem is that with the current API all waits happen sequentially. E.g. if the service of `N` nodes isn't available the CLI will hang for `N * timeout`. I would prefer that the CLI only blocks for `timeout` seconds (whatever that value is). But that is pretty much our of scope for this PR.

Reply:
Please create a follow up ticket for this.

Reply:
Oh, good point. See https://github.com/ros2/ros2cli/issues/507.